Illustration Guideline (#224)
pxblue/doc-it: src/docs/style/illustrations.mdx

import { MaterialDesignDescription, ImageGrid } from '../../app/components';
import { PictureAsPdf as PDFIcon } from '@material-ui/icons';
import LightThemeImage from '../../app/assets/design-patterns/illustrations/illustration-light-theme.svg';
import DarkThemeImage from '../../app/assets/design-patterns/illustrations/illustration-dark-theme.svg';
import BackgroundShapeImage from '../../app/assets/design-patterns/illustrations/illustration-background.png';

<!-- KEYWORDS: design illustration graph onboarding -->

# Illustrations

> Illustrations are design elements that should be used mostly for onboarding screens and [empty states](/patterns/empty-states). Do not use illustrations where you are supposed to use icons.

Reviewer: ```suggestion
Illustrations are bold graphics that are meant to stand out on a page. They should be used primarily for onboarding screens and [empty states](/patterns/empty-states). 

> Illustrations are meant to be large and detailed graphics. Do not use them in places where icons are more appropriate.
```
Will our empty state components support a png/svg instead of an icon?

Author: React takes `ReactNode` 
React Native takes `React.Component<{ size: number, color: string }>`

Reviewer: Okay, React should be fine...RN will probably need an update - want to log an enhancement issue? And Angular is just content projection right? So it can be anything?

Author: Angular: correct. As long as it has `pxb-empty-icon` as the selector

Author: https://github.com/pxblue/react-native-component-library/issues/69

> When creating your own illustrations, you should follow our guidelines as explained below. Alternatively, you may also choose to acquire existing artworks from repositories like [Icons8](https://icons8.com/illustrations), [Adobe Stock](https://stock.adobe.com/vectors), or [Humaaans Figma plugin](https://www.figma.com/community/plugin/739503328703046360). We recommend you finalizing the illustration with [Adobe Illustrator](https://www.adobe.com/products/illustrator.html), as it provides more export options.

Reviewer: ```suggestion
You should adhere to these guidelines when creating your own illustrations. Alternatively, you may also use existing illustrations/artwork from repositories such as [Icons8](https://icons8.com/illustrations), [Adobe Stock](https://stock.adobe.com/vectors), or [Humaaans Figma plugin](https://www.figma.com/community/plugin/739503328703046360). We recommend finalizing your illustrations with [Adobe Illustrator](https://www.adobe.com/products/illustrator.html), as it provides more export options than other tools.
```

<React.Fragment>
    <MaterialDesignDescription
        title={'Onboarding'}
        url={'https://material.io/design/communication/onboarding.html'}
        description={`Material Design's guidelines on onboarding screens.`}
    />
    <MaterialDesignDescription
        title={'Google Marketing Platform Visual Guidelines'}
        url={
            'https://designguidelines.withgoogle.com/ads-branding/assets/1hquKP2EcBlDaWyaLkoezqd8r9KPjRgnp/visual-guidelines-google-marketing-platform.pdf#'
        }

>         description={`You are encouraged to follow their illustration guidelines.`}

Reviewer: Might be nice to mention the page number of the illustration guidelines (they are way down in that document and you have to scan through a lot of irrelevant material to find it).

        avatar={<PDFIcon style={{ width: 40, height: 40 }} />}
    />
</React.Fragment>

## Designing the Illustration

### Perspective

> Your illustration should be either in a horizontal perspective (neither looking up nor looking down) or in parallel projection.

Reviewer: Are these standard design/art terms (horizontal perspective and parallel projection)?

Author: parallel projection: yes

horizontal perspective: I looked it up everywhere but couldn't find a proper way to say it (that term is called "the plain perspective: in Chinese, but it makes no sense here, and I didn't find any translation to that)

Reviewer: Maybe something like "facing the subject matter parallel to the ground (not viewing it from above or below)"

### Ratio

> The width-height ratio should be between 1:1 to 2:3, which gives a margin on a mobile phone for UI interactions. If designing specifically for desktop applications (that is, the illustration will never show up on mobile), you can go wider than that, but no more than 16:9.

Reviewer: ```suggestion
The aspect ratio of your illustrations should be between 1:1 and 2:3. If you are designing exclusively for desktop applications (i.e., the illustration will never show up on mobile), you can go wider, but no more than 16:9.
```

Author: Why taking out the "which gives a margin on a mobile phone for UI interactions"?

Reviewer: what does it mean?

### Drawing Style

> Keep your style consistent within one application, including your illustrations for the webpage for the product. Do not be too abstract or artistic, as your main goal is to inform your users of certain features in your product, instead of decorating it.

Reviewer: ```suggestion
Keep your style consistent within the application. This style should also extend to any marketing sites for the product for consistency. Do not be too abstract or artistic — your main goal is to inform your users of the features in your product, not to decorate it.
```

### Text

> Avoid the use of text, as they might require translations, and can be hard to see on mobile. If you have to, use Open Sans, just like other PX Blue applications, and ensure that they are legit on the smallest targeted device.

Reviewer: ```suggestion
Avoid the use of text wherever possible. It can be hard to see on mobile and will be ineffective if your application needs to support multiple languages. If you have to use text, use Open Sans as the font and ensure that words are legible on your smallest targeted device.
```

### Color

> Use [PX Blue palette](/style/color) as the base color. If there is a gradient, make sure the two ends of the gradient are using PX Blue colors. Use some blue hue (this can be colors from primary blue, Eaton blue, or light blue) somewhere in the foreground for your drawing.

Reviewer: ```suggestion
Use the [PX Blue color palette](/style/color) for your primary colors. You should use some shade of blue in the foreground of your images (these can be colors from the Blue, Light Blue, or Branding Blue palettes).

> When using gradients, both ends of the gradient should be set to colors from the PX Blue palette.
```

Author: When I say "base color", I was suggesting the color before people apply shadows to it (its own color)

Reviewer: main colors? maybe kyle has another word for it?

Author: "local color"? https://en.wikipedia.org/wiki/Local_color_(visual_art)

> Avoid using colors that may blend into the app theme background (suggest adding a background color to pad around it). Avoid using the 500 colors from the primary color palette, as they may blend with other UI elements (buttons, progressors).

Reviewer: ```suggestion
Avoid using colors that may blend into the application background (add a background color to to your images to help them stand out). Avoid using the 500 colors from the color palette, as they may blend in among other UI elements (buttons, etc.).
```

Author: "Avoid using the 500 colors from the **primary** color palette" emphasizes that you can not use blue 500, white 500, gray 500, black 500, and darkBlack 500. All the other 500 colors are acceptable because they are not usually used anywhere in an onboarding screen.

Reviewer: We don't have anything that we call a Primary color palette. We have User Interface colors, Status Colors, and Branding Colors. We could probably say "from the User Interface palettes" and be fine

> Avoid too many dark strokes for light themes, or light strokes for dark themes. The illustrations might be reused as infographics, and line strokes might intervene with labels.

Reviewer: ```suggestion
Avoid using too many strokes, especially light strokes on dark backgrounds or vice-versa. Your illustrations may be reused as infographics, and line strokes could interfere with labels.
```

> Test your illustration in both light theme (white 50) and dark theme (black 900). It is recommended that you modify your design for light backgrounds and dark backgrounds.

Reviewer: ```suggestion
Test your illustrations on both light (white 50) and dark (black 900) application backgrounds. If your application will support multiple themes, we recommended having a version of your icons for each.
```

Author: Modified to "version of your **illustrations** for each"

> Simulate your illustrations under different colorblind conditions. Ensure your color schemes are still intact.

Reviewer: ```suggestion
To maximize accessibility, use online tools (TODO: link to example?) to simulate viewing your illustrations under different colorblind conditions. Ensure that your illustrations are still visible and effective.
```

<ImageGrid
    images={[LightThemeImage, DarkThemeImage]}

>     caption={'Two versions of the same design, under light theme (left) and dark theme (right) respectively.'}

Reviewer: ```suggestion
    caption={'Two versions of the same design for light theme (left) and dark theme (right).'}
```

/>

> <ImageGrid images={[BackgroundShapeImage]} caption={'The illustration background can either be a confined shape (left) or span through the entire container (right).'} gridImageProps={{ sm: 12 }} />

Reviewer: ```suggestion
<ImageGrid images={[BackgroundShapeImage]} caption={'Illustration backgrounds can be either a fixed shape (left) or full-bleed (right).'} gridImageProps={{ sm: 12 }} />
```

## Export

> When finished with your artwork, remember to outline stroke. However, if there are texts in your illustration, do not outline stroke any of that -- leave them in as text so that other people can conveniently translate it later, if necessary.

Reviewer: Is outline stroke an illustrator thing? Will people know what this means?
```suggestion
When finished with your artwork, remember to outline stroke. If you are using text in your illustrations (see above), do not convert to outline or stroke any of those elements — leave them as text for future translation, if necessary.
```

Name and group your root SVG layers semantically for animating. Do not minify the SVG code, as this will eliminate the naming.

> Export your illustration in SVG with at least 2 decimal precision & presentation attributes.

Reviewer: ```suggestion
Export your illustrations as SVG with at least 2 decimal precision and presentation attributes enabled. After exporting, check that your graphics look correct (for higher detail graphics, you may need a higher decimal precision to keep paths smooth).
```

### Ratio

The aspect ratio of your illustrations should be between 1:1 and 2:3 to avoid filling the full screen on mobile devices (leave room for buttons, steppers, etc.). If you are designing exclusively for desktop applications (i.e., the illustration will never show up on mobile), you can go wider, but no more than 16:9.

### Drawing Style

Keep your style consistent within the application. This style should also extend to any marketing sites for the product for consistency. Do not be too abstract or artistic — your main goal is to inform your users of the features in your product, not to decorate it.

### Text

Avoid the use of text wherever possible. It can be hard to see on mobile and will be ineffective if your application needs to support multiple languages. If you have to use text, use Open Sans as the font and ensure that words are legible on your smallest targeted device.

### Color

Use the [PX Blue color palette](/style/color) for your [local colors](<https://en.wikipedia.org/wiki/Local_color_(visual_art)>). You should use some shade of blue in the foreground of your images (these can be colors from the Blue, Light Blue, or Branding Blue palettes).

> Avoid using colors that may blend into the app theme background (suggest adding a background color to pad around it). Avoid using the 500 colors from the from the User Interface palettes, as they may blend with other UI elements (buttons, progressors).

Reviewer: This paragraph is the old version of the one below it...i think we had a merge snafu
```suggestion
```

Author: Thanks for checking :)

Avoid using colors that may blend into the application background (add a background color to to your images to help them stand out). Avoid using the 500 colors from the color palette, as they may blend in among other UI elements (buttons, etc.).

Avoid using too many strokes, especially light strokes on dark backgrounds or vice-versa. Your illustrations may be reused as infographics, and line strokes could interfere with labels.

> Test your illustrations on both light (white 50) and dark (black 900) application backgrounds. If your application will support multiple themes, we recommended having a version of your icons for each.

Reviewer: ```suggestion
Test your illustrations on both light (white 50) and dark (black 900) application backgrounds. If your application will support multiple themes, we recommended having a version of your illustrations for each.
```
